runtime/near-vm-logic/src/logic.rs

```diff
@@ -2391,6 +2389,22 @@ impl<'a> VMLogic<'a> {
         Ok(res? as u64)
     }
 
+    /// Debug print given utf-8 string to node log. It's only available in Sandbox node
+    ///
+    /// # Errors
+    ///
+    /// * If string is not UTF-8 returns `BadUtf8`
+    ///
+    /// # Cost
+    ///
+    /// 0
+    #[cfg(feature = "sandbox")]
+    pub fn debug_log(&mut self, len: u64, ptr: u64) -> Result<()> {
+        let message = self.get_utf8_string(len, ptr)?;
+        tracing::debug!(target: "sandbox", "{}", message);
```

Comment:
I believe this is better written as `tracing::debug!(target: "sandbox", message);`. The primary text of the log message is conventionally represented as the `message` key-value and this is guaranteed to avoid invoking the formatting machinery.

Reply:
Seems doesn't compile:
```
nearcore (sandbox-debug-log) make sandbox
cargo build -p neard --features sandbox
   Compiling near-vm-logic v0.0.0 (/home/bo/workspace/nearcore/runtime/near-vm-logic)
error[E0277]: the trait bound `std::string::String: tracing::Value` is not satisfied
    --> runtime/near-vm-logic/src/logic.rs:2417:9
     |
2417 |         tracing::debug!(target: "sandbox", message);
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tracing::Value` is not implemented for `std::string::String`
     |
     = note: required for the cast to the object type `dyn tracing::Value`
     = note: this error originates in the macro `$crate::valueset` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `near-vm-logic` due to previous error
make: *** [Makefile:82: neard-sandbox] Error 101
```

Reply:
Ah, I missed this is a `String`. You'll need an `&str`, [like this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3f556a015b53c73b8d87cb8b218fbdf9).

Reply:
Got it, thanks! I learned more syntax about it on tracing's docs page. I also need a `message =`.